src/gasman.c

```diff
@@ -1330,14 +1330,17 @@ void            RetypeBag (
     UInt                new_type )
 {
     BagHeader * header = BAG_HEADER(bag);
+    UInt        old_type = header->type;
```

Comment:
Is this check worth doing, as all we do is assign header->type=new_type?

Reply:
It's irrelevant in single threaded mode. In multi-threaded code on multiple cores, it can invalidate cache lines for no good reason and thus degrade performance.

Reply:
At first I thought gcc/clang would optimise this away, but on further thought they aren't allowed to (as the memory could be read-only). It's fine anyway.